### DISCUSSION TITLE
How to change the color of the filename in grep_project (with rg)

### DISCUSSION THREAD

**Discussion by skamsie** [ibhagwan/fzf-lua#1516] [Q&A] [answered]
I am trying to change the color of the filename for the `grep_project` command. It is currently violet, see screenshot below. I have fiddled with all the `FzfLua...` highlight groups I could find and with the `fzf_colors = {...` but nothing seems to affect it. I then thought it's set by `rg` itself, but if I run the command in the terminal it's a different color (pink), so something in `nvim` must set it to violet...

<img width="1040" alt="Screenshot 2024-11-06 at 10 15 07 PM" src="https://github.com/user-attachments/assets/0cec099d-7ec2-44ff-a3a2-7ac51cee43c8">

**Comment by ibhagwan** [comment]
That’s the color output from rg.

**Comment by ibhagwan** [comment]
If you use a formatted fzf-lua will strip the ansi highlight and can apply a new one, for example:
```lua
:FzfLua grep_project formatter=path.dirname_first hls.file_part=Error hls.dir_part=Direcrory
```
![IMG_2669](https://github.com/user-attachments/assets/560f085e-29cb-4346-a985-b280e4b74031)

**Reply by skamsie** [reply]
Thanks, this works! Do you know what would be the syntax to add the colors somewhere in the config? This is how it currently looks:

```lua
-- Improved fzf.vim written in lua
return {
  "ibhagwan/fzf-lua",
  config = function()
    local function set_fzf_keymap(key, func)
      vim.keymap.set(
        'n', '<leader>' .. key,
        function()
          require('fzf-lua')[func]()
        end,
        { noremap = true, silent = true }
      )
    end

    set_fzf_keymap('f', 'files')
    set_fzf_keymap('a', 'grep_project')
    set_fzf_keymap('h', 'oldfiles')
    set_fzf_keymap('b', 'buffers')

    -- calling `setup` is optional for customization
    local actions = require 'fzf-lua.actions'
    require("fzf-lua").setup(
      {
        winopts = {
          row = 0.5,
          col = 0.5,
          width = 0.8,
          height = 0.8,
          border = 'single',
          prompt = 'B',
          backdrop = 100
        },
        fzf_opts = { ['--layout'] = 'default' },
        defaults = { file_icons = false },
        files = {
          winopts = { preview = { delay = 10 } }
        },
        oldfiles = {
          winopts = { preview = { delay = 10 } }
        },
        grep = {
          actions = {
            ['ctrl-g'] =  actions.toggle_ignore,
          }
        }
      }
    )
  end
}
```

**Reply by ibhagwan** [reply, marked as answer]
Add it either under grep or under defaults (if you want it to apply to all pickers):
```lua
 grep = {
    formatter = 'path.dirname_first',
    hls = {
      dir_part = 'Directory',
      file_part = …
    }
 }
```

**Reply by skamsie** [reply]
thank you! One follow-up question though... based on your example and the documentation I tried to also change the color of the linenumber which I assumed would be `path_linenr` but it had no effect (still green). Is there another name for that one?

```lua
hls = {
  dir_part = 'Normal',
  file_part = 'Normal',
  path_linenr = 'Normal',
  cursorlinenr = 'Normal'
}
```

<img width="838" alt="Screenshot 2024-11-07 at 12 22 39 AM" src="https://github.com/user-attachments/assets/72fcce9d-de52-4463-bc86-3ca341a9de4a">

**Reply by ibhagwan** [reply]
> thank you! One follow-up question though... based on your example and the documentation I tried to also change the color of the linenumber which I assumed would be `path_linenr` but it had no effect (still green). Is there another name for that one?
> 
> ```lua
> hls = {
>   dir_part = 'Normal',
>   file_part = 'Normal',
>   path_linenr = 'Normal',
>   cursorlinenr = 'Normal'
> }
> ```

These might need to be setup globally (not under defaults), alternatively you can link the `FzfLuaDirPart` as explained here:
https://github.com/ibhagwan/fzf-lua?tab=readme-ov-file#highlights

**Reply by skamsie** [reply]
I tried, but unfortunately neither had any effect. Maybe a bug?

```lua
    require("fzf-lua").setup(
    {
      hls = {
        dir_part = 'Normal',
        file_part = 'Normal',
        path_linenr = 'Normal',
        cursorlinenr = 'Normal'
      },
```

**Reply by ibhagwan** [reply]
> I tried, but unfortunately neither had any effect. Maybe a bug?
> 
> ```lua
>     require("fzf-lua").setup(
>     {
>       hls = {
>         dir_part = 'Normal',
>         file_part = 'Normal',
>         path_linenr = 'Normal',
>         cursorlinenr = 'Normal'
>       },
> ```

It works in the command I originally posted but not with `setup`?
When you say if doesn’t work what do you mean? Is it still colored?
Where did you add the formatter part (should be either in the command or under `defaults`)?

**Reply by skamsie** [reply]
@ibhagwan I am talking specifically about not being able to change the color of the line number (I assume it should be `path_linenr`). No matter what I set it to, it's still the default green. Even if I put it in setup directly or in `defaults`. Everything else works as intended

**Reply by ibhagwan** [reply]
> @ibhagwan I am talking specifically about not being able to change the color of the line number (I assume it should be `path_linenr`). No matter what I set it to, it's still the default green. Even if I put it in setup directly or in `defaults`. Everything else works as intended

Line number is set by rg not fzf-lua, the highlight only takes effect in pickers where the output is created by fzf-lua (buffers, LSP, etc), you can try changing terminal green in your neovim color scheme.
